kafkabp/consumer.go

```diff
@@ -214,10 +262,24 @@ func (kc *consumer) Consume(
 	for {
 		// create a partition consumer for each partition
 		consumer := kc.getConsumer()
-		partitions := *kc.partitions.Load()
-		partitionConsumers := make([]sarama.PartitionConsumer, 0, len(partitions))
+		partitions := kc.getPartitions()
+		numPartitions := len(partitions)
+		partitionConsumers := make([]sarama.PartitionConsumer, 0, numPartitions)
+		partitionsToConsume := make([]bool, numPartitions)
+		partitionsSelector := kc.partitionSelectorProvider(numPartitions)
+		if partitionsSelector == nil {
+			return ErrNilPartitionSelector
+		}
+		for i := 0; i < numPartitions; i++ {
+			partition := partitions[i]
+			partitionsToConsume[partition] = partitionsSelector(partition)
+		}
```

Comment:
```suggestion
		for _, partition := range partitions {
			partitionsToConsume[partition] = partitionsSelector(partition)
		}
```

Maybe easier to read since we don't need the slice indexing here?